ecosystem/sep-0006.md

```diff
@@ -30,6 +30,73 @@ This proposal defines a standard protocol enabling the following features direct
 
 To support this protocol an anchor acts as a server and implements the specified REST API endpoints, while a wallet implements a client that consumes the API. The goal is interoperability, so a wallet implements a single client according to the protocol, and will be able to interact with any compliant anchor. Similarly, an anchor that implements the API endpoints according to the protocol will work with any compliant wallet.
 
+## Diagrams
+
+The diagrams provided offer a detailed view of various flows that are possible using this protocol. Note that these diagrams are opinionated. Variations of the these flows may also be sufficient.
+
+### Completing a Withdrawal Transaction with a Firm Quote
+
+This diagram demonstrates the interactions between the entities involved in a successful transaction. Specifically it uses the [Asset Conversions Using a Firm Quote](#asset-conversions-using-a-firm-quote) strategy.
+
+```mermaid
+sequenceDiagram 
+    participant User
+    participant Wallet
+    participant Stellar
+    participant Anchor
+    User->>Wallet: initiates withdrawal to <br> User's country
+    Wallet-->>Wallet: maps country to Anchor
+
+    Wallet->>+Anchor: GET /.well-known/stellar.toml
+    Anchor-->>-Wallet: SEP-10, 12, 6, & 38 URLs
+    Wallet->>+Anchor: GET [SEP-6]/info
+    Anchor-->>-Wallet: Stellar assets, customer types, fees
+    Wallet->>+Anchor: GET [SEP-38]/prices
+    Anchor-->>-Wallet: pairs and est. rates for Stellar asset
+
+    Wallet->>+User: provides pair <br> est. rates & fees
+    User-->>-Wallet: selects asset pair
+
+    Wallet->>+Anchor: GET [SEP-10]
+    Anchor-->>-Wallet: challenge transaction
+    Wallet-->>Wallet: signs challenge
+    Wallet->>+Anchor: POST [SEP-10]
+    Anchor-->>Anchor: verifies challenge
+    Anchor-->>-Wallet: authentication token
+
+    
+    note right of Wallet: If User has already <br> been accepted, <br> KYC is unnecessary.
+    Wallet->>+Anchor: GET [SEP-12]/customer?type=
+    Anchor-->>-Wallet: fields required
+    Wallet->>+User: requests fields
+    User-->>-Wallet: provides field values
+    Wallet->>+Anchor: PUT [SEP-12]/customer?type=
+    Anchor-->>Anchor: validates KYC values
+    Anchor-->>-Wallet: id, ACCEPTED
+
+    Wallet->>+Anchor: GET [SEP-38]/price
+    Anchor-->>-Wallet: exchange rate
+    Wallet->>+User: provides estimated rate
+    User-->>-Wallet: continues
+    Wallet->>+Anchor: POST [SEP-38]/quote
+    Anchor-->>Wallet: quote id, rate, expiration
+    
+    Wallet->>+Anchor: GEt [SEP-6]/withdraw
+    Anchor-->>Anchor: checks customer statuses, <br> links quote, <br> creates transaction record
+    Anchor-->>-Wallet: transaction id, receiving account & memo
+
+    Wallet->>+Stellar: submit Stellar payment
+    Stellar->>+Anchor: receives payment, matches w/ transaction
+    Anchor-->>Anchor: updates transaction status
+    Stellar-->>-Wallet: success response
```

Comment:
❔ It feels to me like the two actions/events involving the anchor here may be too early in the flow. The anchor is unlikely to receive the payment until after the transaction succeeds, which his illustrated just below these steps on the next line. 

This is what it looks like now:
<img width="554" alt="Screen Shot 2022-03-17 at 9 22 00 AM" src="https://user-images.githubusercontent.com/351529/158846441-a2f90a37-d033-4f80-bd3f-478918aaba4d.png">

It might be worth disconnecting the two like this:
```suggestion
    Wallet->>+Stellar: submit Stellar payment
    Stellar-->>-Wallet: success response
    Stellar->>+Anchor: receives payment, matches w/ transaction
    Anchor-->>Anchor: updates transaction status
```
<img width="554" alt="Screen Shot 2022-03-17 at 9 21 51 AM" src="https://user-images.githubusercontent.com/351529/158846532-dabe35d0-658a-4723-88d6-619b24f9bcbb.png">

Or, I realize that these are not sequentially ordered processes and are happening in parallel, and so you could represent this as a parallel block. This might complicate the diagram a little too much though.
```suggestion
    Wallet->>+Stellar: submit Stellar payment
    par
        Stellar-->>-Wallet: success response
    and
        Stellar->>+Anchor: observes payment, matches w/ transaction
        Anchor-->>Anchor: updates transaction status
    end
```
<img width="554" alt="Screen Shot 2022-03-17 at 9 21 21 AM" src="https://user-images.githubusercontent.com/351529/158846614-ba1d38f3-d820-43fa-b666-15b709deb7ff.png">



I don't feel strongly about any of these changes and are mostly thinking out loud. Defer to you if you think any of the above is a good idea and helps or not.